indexing-service/src/main/java/io/druid/indexing/common/actions/SegmentListUsedAction.java

```diff
@@ -53,9 +69,9 @@ public String getDataSource()
   }
 
   @JsonProperty
```

Comment:
during a typical ugprade, old middlemanagers will keep running tasks for many hours after the overlord has been updated. Even if this method is only called at the start of a task, there is a possible race condition if the overlord gets updated right after assigning a task to an old node, causing this task to fail when it tries to use this task action on the updated overlord.